service-tests/json.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+const Joi = require('joi');
+const ServiceTester = require('./runner/service-tester');
+
+const t = new ServiceTester({ id: 'json', title: 'User Defined JSON Source Data' });
+module.exports = t;
+
+t.create('Connection error')
+  .get('/Package Name/$.name/https://github.com/badges/shields/raw/master/package.json.json')
+  .networkOff()
+  .expectJSON({ name: 'Package Name', value: 'inaccessible' });
+
+t.create('JSON from url')
+  .get('/Package Name/$.name/https://github.com/badges/shields/raw/master/package.json.json')
+  .expectJSON({ name: 'Package Name', value: 'gh-badges'});
+
+t.create('JSON from url | w/prefix & suffix')
+  .get('/Shields-3498db-v- dev/$.version/https://github.com/badges/shields/raw/master/package.json.json')
```

Comment:
I know we use these path components with dashes in the website badge, though in my experience they are a bit difficult to make sense of. There are also some limitations: they require custom code to escape them (as your code in try.html shows.

What do you think about handling these extra options in the querystring, leaving just the URL in the URL? Since query strings provide named parameters it's easier to decode what is what.

Reply:
Yeah sounds like a good plan, will work on updating to that in the next couple of days

so the url for it would be `shields.io/json/https://website.com/pathtomyjson.svg`?
with all the other parts being in the query string

`label`
`prefix`
`suffix`
`colorB`
**`data`** for the JSONpath/XMLpath? as could use the same query param for the static badge possibly?

Reply:
Thanks for your energy to push this forward! It's a really powerful feature and I'm stoked to add it, and also think it's important to get it right.

`label` is also handled by `getBadgeData` btw.

`data` is not so descriptive. How about `query`?

The URL you're showing looks good, though let's make sure the scheme (http or https) is handled the same as the website badge.

Reply:
Cool removed the `query.label` part.

Yeah query works.

so show the URL like `/badge/dynamic/json/https/github.com` instead of `/badge/dynamic/json/https://github.com`?

or should the URL be apart of the query params as well?
which would make the URL for the badge something like:
`//shields.io/badge/dynamic/json.svg?url=https://github.com/pathtojson.json`
`//shields.io/badge/dynamic/json.svg?datasource=https://github.com/pathtojson.json`